Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -109,7 +109,22 @@ void SoftwareBreakpointManager::enumerate(
 bool SoftwareBreakpointManager::hit(Target::Thread *thread) {
   ds2::Architecture::CPUState state;
   thread->readCPUState(state);
-  return super::hit(state.pc());
+  state.setPC(state.pc() - 2);
```

Comment:
I think this whole block should happen only `#if defined(OS_WIN32)`. Linux seems to handle the PC adjustment by itself.
If it's not too annoying, can you test that behavior yourself?